### PR TITLE
Make "Extended Vector" Default

### PR DESCRIPTION
**Pull request by VinInn**
Following the presentations and discussion at the last 2 XCs
( see https://indico.cern.ch/event/1406814/#2-inputs-from-software-and-rec )
I propose here to make "ExtVec" default.

This choice is more sustainable in the long term as uses a single code-base for all platforms and versions (but ICC compiler).

I've also (re-)introduced fast in AnalyticJacobians as tests show a clear benefit.

regression expected (regression triggered the discussion after all!)

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild : please test

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@cmsbuild, please test for el8_aarch64_gcc12

**Comment by VinInn** [comment]
Apparently never compiled with clang!

**Comment by VinInn** [comment]
@cmsbuild : please test

**Comment by VinInn** [comment]
@cmsbuild, please test for el8_aarch64_gcc12

**Comment by mandrenguyen** [comment]
+1

**Comment by slava77** [comment]
before merging it would still be good to see the standard build results with comparisons, since changes are expected

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
> [Reco comparison results: 24143 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-25-1100+71a331/62561/validateJR.html)

 numerous diffs, but looking at a few workflows tracking-related quantities, the differences are consistent with what usually shows up from just numerical precision changes

**Comment by antoniovilela** [comment]
@cms-sw/reconstruction-l2 
Differences in tests look ok?

**Comment by rappoccio** [comment]
+1

File: DataFormats/Math/interface/SIMDVec.h [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+//
+//
 #if (defined(__CLING__) || defined(__MIC__) || defined(__NVCC__)) || (__BIGGEST_ALIGNMENT__ < 16)
-#elif defined(__GNUC__) || defined(__clang__) || defined(__INTEL_COMPILER)
-#if defined(__x86_64__) && defined(__SSE__)
+#elif defined(__INTEL_COMPILER)
+// intel compiler does not support the extended vector syntax
+#define USE_SSEVECT
+#elif defined(__GNUC__) || defined(__clang__)
+#if defined(__x86_64__) && defined(__SSE__) && defined(CMS_PREFER_SSEVECT)
```
Comment (smuzaffar) [inline DataFormats/Math/interface/SIMDVec.h:13]: @VinInn , looks like this broke CLANG builds. Before this change, for `clang` we were using/defining  `USE_SSEVECT` but now as `CMS_PREFER_SSEVECT` is not defined so for clang we end up using `USE_EXTVECT`

**Comment by VinInn** [comment]
sort of "expected"
will fix today.
is there any  gcc IB I can start from that include it?

**Comment by iarspider** [comment]
> sort of "expected" will fix today. is there any gcc IB I can start from that include it?

Clang IB from 23h yesterday

**Comment by VinInn** [comment]
fixed in #44885

**Pull request merged**